CMakeLists.txt

```diff
@@ -344,24 +343,53 @@ foreach(subtarget ${subtargets})
           ARCHIVE DESTINATION ${LIBRARY_PATH})
 endforeach ()
 
-add_executable (bench ${fftw_libbench2_SOURCE} tests/bench.c tests/hook.c tests/fftw-bench.c)
-
-if (ENABLE_THREADS AND NOT WITH_COMBINED_THREADS)
-  target_link_libraries (bench ${fftw3_lib}_threads)
-else ()
-  target_link_libraries (bench ${fftw3_lib})
+install (FILES api/fftw3.h DESTINATION include RENAME fftw3${PREC_SUFFIX}.h)
+if (EXISTS api/fftw3.f)
+  install (FILES api/fftw3.f DESTINATION include RENAME fftw3${PREC_SUFFIX}.f)
 endif ()
 
+if (BUILD_TESTS)
+
+  add_executable (bench ${fftw_libbench2_SOURCE} tests/bench.c tests/hook.c tests/fftw-bench.c)
 
-enable_testing ()
+  if (ENABLE_THREADS AND NOT WITH_COMBINED_THREADS)
+    target_link_libraries (bench ${fftw3_lib}_threads)
+  else ()
+    target_link_libraries (bench ${fftw3_lib})
+  endif ()
 
-if (Threads_FOUND AND BUILD_TESTS)
 
-  macro (fftw_add_test problem)
-    add_test (NAME ${problem} COMMAND bench -s ${problem})
-  endmacro ()
+  enable_testing ()
 
-  fftw_add_test (32x64)
-  fftw_add_test (ib256)
+  if (Threads_FOUND)
 
+    macro (fftw_add_test problem)
+      add_test (NAME ${problem} COMMAND bench -s ${problem})
+    endmacro ()
+
+    fftw_add_test (32x64)
+    fftw_add_test (ib256)
+
+  endif ()
 endif ()
+
+# pkgconfig file
+set (prefix ${CMAKE_INSTALL_PREFIX})
+set (exec_prefix ${CMAKE_INSTALL_PREFIX})
+set (libdir ${CMAKE_INSTALL_PREFIX}/${LIBRARY_PATH})
+set (includedir ${CMAKE_INSTALL_PREFIX}/include)
+set (VERSION ${FFTW_VERSION})
+configure_file (fftw.pc.in fftw${PREC_SUFFIX}.pc @ONLY)
+install (FILES
+          ${CMAKE_CURRENT_BINARY_DIR}/fftw${PREC_SUFFIX}.pc
+         DESTINATION ${LIBRARY_PATH}/pkgconfig
+         COMPONENT Development)
+
+# cmake file
+configure_file (FFTW3Config.cmake.in FFTW3${PREC_SUFFIX}Config.cmake @ONLY)
+configure_file (FFTW3ConfigVersion.cmake.in FFTW3${PREC_SUFFIX}ConfigVersion.cmake @ONLY)
+install (FILES
+          ${CMAKE_CURRENT_BINARY_DIR}/FFTW3${PREC_SUFFIX}Config.cmake
+          ${CMAKE_CURRENT_BINARY_DIR}/FFTW3${PREC_SUFFIX}ConfigVersion.cmake
+         DESTINATION ${LIBRARY_PATH}/cmake/fftw3
```

Comment:
Instead of `${LIBRARY_PATH}` use `include(GNUInstallDirs)` and `CMAKE_INSTALL_LIBDIR` will have the right value automatically.